llvm/lib/Target/AIE/AIE2PreLegalizerCombiner.cpp

```diff
@@ -177,6 +182,81 @@ bool AIE2PreLegalizerCombinerImpl::tryToCombineSetExtract(
   return true;
 }
 
+// Returns a map with InsertIndices and registers holding the insert values.
+std::optional<std::map<unsigned, Register>>
```

Comment:
I feel we could simplify that return type. How about just `std::map<unsigned, Register>`? The returned map would be empty if we did not manage to gather all indices.